gamemodes/terrortown/gamemode/server/sv_weapon_pickup.lua

```diff
@@ -0,0 +1,32 @@
+util.AddNetworkString("ttt2_switch_weapon")
+
+net.Receive("ttt2_switch_weapon", function(_, ply)
+	-- player and wepaon must be valid
+	if not IsValid(ply) or not ply:IsTerror() or not ply:Alive() then return end
+
+	-- do not allow weapon switch when the weapon switch was started in the previous round
+	local roundstate_client = net.ReadString()
+	if roundstate_client == ROUND_POST and roundstate_client ~= GetRoundState() then return end
```

Comment:
Is there a reason for this being sent by the client? Why do we not use the serverside roundstate?